components/style/values/generics/svg.rs

```diff
@@ -101,80 +102,68 @@ impl<ColorType: Parse, UrlPaintServer: Parse> Parse for SVGPaint<ColorType, UrlP
 /// A value of <length> | <percentage> | <number> for svg which allow unitless length.
 /// https://www.w3.org/TR/SVG11/painting.html#StrokeProperties
 #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
-#[derive(Clone, Copy, Debug, HasViewportPercentage, PartialEq, ToAnimatedValue)]
-#[derive(ToAnimatedZero, ToCss, ToComputedValue)]
-pub enum SvgLengthOrPercentageOrNumber<LengthOrPercentage, Number> {
+#[derive(Clone, Copy, Debug, HasViewportPercentage, PartialEq, ToCss)]
+#[derive(ToComputedValue)]
+pub enum SvgLengthOrPercentageOrNumber<L, N> {
     /// <length> | <percentage>
-    LengthOrPercentage(LengthOrPercentage),
+    LengthOrPercentage(L),
     /// <number>
-    Number(Number),
+    Number(N),
 }
 
-impl<L, N> ComputeSquaredDistance for SvgLengthOrPercentageOrNumber<L, N>
+/// Following From implements use for converting animated value.
+impl<L, N> ToAnimatedValue for SvgLengthOrPercentageOrNumber<L, N>
     where
-        L: ComputeSquaredDistance + Copy + Into<NumberOrPercentage>,
-        N: ComputeSquaredDistance + Copy + Into<NumberOrPercentage>
+        L: Into<SvgStrokeNumberOrPercentageOrCalc> + From<LengthOrPercentage>,
+        N: Into<SvgStrokeNumberOrPercentageOrCalc> + From<Number>
 {
+    type AnimatedValue = SvgStrokeNumberOrPercentageOrCalc;
```

Comment:
No, something named "length" shouldn't have a more specific animated value with "stroke" in its name. What exactly are you doing with this patch?

Reply:
Thanks for reviewing.
I addressed the patch:
 * Rename SvgStrokeNumberOrPercentageOrCalc to SvgNumberOrPercentageOrCalc.
 * Use ToAnimatedValue derive.